[Jeromy Tan] iP

jxromy opened the pull request.

jxromy committed:
Rename, Greet, Exit

jxromy committed:
Update README.md

jxromy committed:
Updated Chai to echo inputs

jxromy committed:
Merge branch 'master' of https://github.com/jxromy/ip

jxromy committed:
Updated Chai with List function

jxromy committed:
Update README.md

jxromy committed:
Refactored codebase to observe OOP principles

ChaiBot contains chatbot logic
TaskList manages tasks
UI centralises print logic

jxromy committed:
Added function to mark and unmark tasks

Created Task class with toString that overrides superclass method

TayGuangSheng commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,44 @@
+public class Ui {
```
The class names are well-named for the rest, but consider making UI more descriptive, like User Interface.

jxromy replied:
Thanks for this! I refactored Ui to UserInterface

TayGuangSheng commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+public class Task {
```
Your code follows a clear structure, with each class handling a distinct responsibility.

TayGuangSheng commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,44 @@
+        System.out.println(LINE + "\nError: Please enter a valid task number!\n" + LINE);
+    }
+
+    public static void showOutOfRangeMessage() {
```
Your code handles invalid inputs gracefully, instead of just crashing. However, consider letting the user know what is the valid task number range in this case.

jxromy replied:
Sounds good, I will add the valid inputs in the print statement.

TayGuangSheng commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+public class TaskList {
+    private static final int MAX_SIZE = 100;
```
Instead of using hard-coded numbers, you use constants like MAX_SIZE which is good.

TayGuangSheng reviewed (commented):
Overall, your code is well-structured, readable, and robust.

icknee commented on src/main/java/ChaiBot.java:
```diff
@@ -0,0 +1,37 @@
+            try {
+                int taskNumber = Integer.parseInt(input.substring(5));
+                tasks.markTask(taskNumber);
+            } catch (NumberFormatException e) {
+                Ui.showInvalidNumberMessage();
```
Nice that you handle edge cases

icknee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+        this.isDone = false;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsNotDone() {
+        this.isDone = false;
+    }
```
i think that this can be handled in one method

icknee commented on src/main/java/ChaiBot.java:
```diff
@@ -0,0 +1,37 @@
+        this.tasks = new TaskList();
+    }
+
+    public boolean handleInput(String input) {
```
the if else is structured clearly, easy to understand

icknee commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,44 @@
+public class Ui {
```
Class name is suitable

icknee commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+public class TaskList {
```
Naming for the methods are well done

icknee reviewed (commented):
Good job! Code is very readable, keep up the good work

louistaii commented on src/main/java/ChaiBot.java:
```diff
@@ -1,43 +1,37 @@
+        } else if (input.equals("list")) {
+            tasks.printTasks();
+        } else if (input.startsWith("mark ")) {
+            try {
```
The formatting for the try-catch statements are good!

louistaii commented on src/main/java/Chai.java:
```diff
@@ -1,6 +1,28 @@
+
+        Ui.showWelcomeMessage(logo);
+
+        boolean running = true;
```
Consider renaming this to isRunning

jxromy replied:
Thanks for catching this! I refactored running to to isRunning

louistaii commented on src/main/java/ChaiBot.java:
```diff
@@ -1,43 +1,37 @@
+    public boolean handleInput(String input) {
+        if (input.equals("bye")) {
+            Ui.showExitMessage();
+            return false;
```
Consider changing else if to if statement instead since initial if statement already returns and exits method. No need for else.

louistaii commented on src/main/java/Ui.java:
```diff
@@ -14,13 +14,30 @@ public static void showAddedMessage(String message) {
+        System.out.println(LINE + "\nError: Task number out of range!\n" + LINE);
+    }
+
+    public static void showTaskMarkedMessage(Task task) {
```
Naming of methods are informative and follow the naming conventions. Good job!

louistaii reviewed (commented):
Overall very well written code, with strong organization and adherence to coding conventions.

jxromy committed:
Added ToDos, Events, Deadlines functionality

Minor refactoring done to improve code readability
Implemented changes from code review

jxromy committed:
Added files to semi-automate text UI testing

jxromy committed:
ChaiBot is now implemented with exception handling for invalid inputs

Added ChaiException Class
User is given an explanation of input errors and how to correct it

jxromy committed:
Created specific error handlers in ChaiException

Separated error-checking and exception handling
Defined format error messages in ChaiBot as constants
Improved code readability and modularity

jxromy committed:
Merge branch 'branch-Level-5'

* branch-Level-5:
  Created specific error handlers in ChaiException
  ChaiBot is now implemented with exception handling for invalid inputs

jxromy committed:
Refactoring of user interactions

User is now notified when marking a completed task or unmarking an incomplete task
Chai's logo is now defined as a constant in UserInterface

jxromy committed:
Reorganised classes into java packages

Updated runtest with package destinations

jxromy committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Reorganised classes into java packages

jxromy committed:
Abstracted 9 lines of duplicated code into a helper function

Helper function uses Java Collection class ArrayList methods

jxromy committed:
Added support for deleting tasks from the list

jxromy committed:
Minor improvement to code readability

Added curly braces to all cases in switch-case

jxromy committed:
Added isDone parameter to Task classes

jxromy committed:
Tasks are now saved in the hard disk automatically whenever the task list changes

Data is loaded when the chatbot starts up

jxromy committed:
Merge branch 'branch-Level-6'

* branch-Level-6:
  Minor improvement to code readability
  Added support for deleting tasks from the list
  Abstracted 9 lines of duplicated code into a helper function

jxromy committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Tasks are now saved in the hard disk automatically whenever the task list changes
  Added isDone parameter to Task classes

# Conflicts:
#	src/main/java/chai/tasks/TaskList.java

jxromy committed:
Refactored ChaiBot class as Parser class
Improved OOP structure, main function with only 1 LOC
Split loadTasks function in Storage class to avoid nested code

jxromy committed:
Improved UI with friendlier interactions

jxromy committed:
Removed exceptions throwing in main class

jxromy committed:
Added missing save to storage after task deletion

added new kaomojis to user interactions

jxromy committed:
Chai now understands date and time
System time is displayed upon launch
User is allowed 32 different formats to enter date and time
Refactored constant variables to follow conventions

aditichadha1310 commented on src/main/java/chai/tasks/Event.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public String toString() {
+        return "[" + getTaskType() + "]" + getStatusIcon() + " " + description +
```
Avoid the use of magic strings as much as possible.

aditichadha1310 commented on src/main/java/chai/tasks/Parser.java:
```diff
@@ -0,0 +1,78 @@
+        this.tasks = tasks;
+    }
+
+    public boolean parse(String input) throws ChaiException {
```
Avoid lengthy methods. try to keep the method length to about 30 lines of code,

aditichadha1310 reviewed (commented):
overall, good work!

jxromy committed:
Added find function for users to search for keywords in task description

jxromy committed:
Merge pull request #1 from jxromy/branch-Level-8

Chai now understands date and time

jxromy committed:
Merge pull request #2 from jxromy/branch-Level-9

Added find function for users to search for keywords in task description

jxromy committed:
Added JavaDoc comments

jxromy committed:
Merge branch 'master' into branch-A-JavaDoc

jxromy committed:
Merge pull request #3 from jxromy/branch-A-JavaDoc

Added JavaDoc comments

jxromy committed:
Updated Chai's User Guide

jxromy committed:
Fixed Javadoc comments

Corrected syntax errors after last merge
Added comments for new methods

jxromy committed:
Fixed empty list returned in find function when no matches are found
